probers/filesystems/probe.go

```diff
@@ -92,18 +92,19 @@ func (p *prober) processMountLine(line string) error {
 	if fs.probed {
 		return nil
 	}
-	if fd, err := syscall.Open(mountPoint, syscall.O_RDONLY, 0); err != nil {
+	fd, err := syscall.Open(mountPoint, syscall.O_RDONLY, 0)
+	if err != nil {
 		return errors.New(fmt.Sprintf("error opening: %s %s", mountPoint, err))
-	} else {
-		defer syscall.Close(fd)
```

Comment:
Yes, I see the leak in probers/url/probe.go
I don't see the leak in probers/filesystems/probe.go so I don't see why that code is being changed. In case you're wondering why the code is structured this way rather than something similar to what you changed, the original limits the scope of the fd and err variables, which is generally good practice.